src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        if (typeof action.extraData === 'object' && action.extraData !== null) {
+          Object.assign(currentState, action.extraData);
+        }
```

Comment:
The check for `action.extraData` being an object is good, but it might be more robust to also check that it is not an array, as arrays are also objects in JavaScript. You can use `!Array.isArray(action.extraData)` for this check.